homeassistant/components/history/__init__.py

```diff
@@ -286,6 +329,26 @@ def _get_states_with_session(
     return [LazyState(row) for row in execute(query)]
```

Comment:
I notice we're creating a list here. Back in the day this used to be a generator I think? (which is why we wrap the results in `list(…)` in `get_state()` (which really should be using `next()`))

Reply:
Oh this comment was meant for the same code in `_get_single_entity_states_with_session`

Reply:
I made them generators.

Reply:
I ended up reverting the switch to generators since it didn't improve performance, and the overhead of the additional function calls is likely slower.

Reply:
The `list()` wrapping can come out though since `get_states` always returns a list anyways